src/ol/Image.js

```diff
@@ -22,159 +22,194 @@ import {listenOnce, unlistenByKey} from './events.js';
  * post requests or - in general - through XHR requests, where the src of the
  * image element would be set to a data URI when the content is loaded.
  *
- * @typedef {function(ImageWrapper, string): void} LoadFunction
+ * @typedef {function(import("./Image.js").default, string): void} LoadFunction
  * @api
  */
 
-class ImageWrapper extends ImageBase {
+/**
+ * @typedef {Object} ImageObject
+ * @property {import("./extent.js").Extent} [extent] Extent.
+ * @property {number|Array<number>} [resolution] Resolution, if different from the requested one.
+ * When x and y resolution are different, use the array type (`[xResolution, yResolution]`).
+ * @property {number} [pixelRatio] Pixel ratio, if different from the requested one.
+ * @property {import('./DataTile.js').ImageLike} image Image.
+ */
+
+/**
+ * Loader function used for image sources. Receives extent, resolution and pixel ratio as arguments.
+ * The function returns an {@link import("./DataTile.js").ImageLike image}, an
+ * {@link import("./Image.js").ImageObject image object}, or a promise for the same.
+ * For loaders that generate images, the promise should not resolve until the image is loaded.
+ * If the returned image does not match the extent, resolution or pixel ratio passed to the loader,
+ * it has to return an {@link import("./Image.js").ImageObject image object} with the `image` and the
+ * correct `extent`, `resolution` and `pixelRatio`.
+ *
+ * @typedef {function(import("./extent.js").Extent, number, number, (function(HTMLImageElement, string): void)=): import("./DataTile.js").ImageLike|ImageObject|Promise<import("./DataTile.js").ImageLike|ImageObject>} Loader
+ * @api
+ */
+
+/**
+ * Loader function used for image sources. Receives extent, resolution and pixel ratio as arguments.
+ * The function returns a promise for an  {@link import("./Image.js").ImageObject image object}.
+ *
+ * @typedef {function(import("./extent.js").Extent, number, number, (function(HTMLImageElement, string): void)=): import("./DataTile.js").ImageLike|ImageObject|Promise<import("./DataTile.js").ImageLike|ImageObject>} ImageObjectPromiseLoader
+ */
```

Comment:
Returning an `ImageObject` object literal with the relevant image properties (image, extent, resolution and pixel ratio) rather than an `ImageWrapper` instance is convenient, and should also help avoid headache for 3rd party plugin developers, because they do not need an `ol` dependency when creating loaders.